commons/utils.js

```diff
@@ -66,6 +67,115 @@ const getUnit = bridgeMode => {
   return { unitHome, unitForeign }
 }
 
+const parseValidatorEvent = event => {
+  if (
+    event.event === undefined &&
+    event.raw &&
+    event.raw.topics &&
+    (event.raw.topics[0] === '0xe366c1c0452ed8eec96861e9e54141ebff23c9ec89fe27b996b45f5ec3884987' ||
+      event.raw.topics[0] === '0x8064a302796c89446a96d63470b5b036212da26bd2debe5bec73e0170a9a5e83')
+  ) {
+    const rawAddress = event.raw.topics.length > 1 ? event.raw.topics[1] : event.raw.data
+    const address = '0x' + rawAddress.slice(26)
+    event.event = 'ValidatorAdded'
+    event.returnValues.validator = address
+  } else if (
+    event.event === undefined &&
+    event.raw &&
+    event.raw.topics &&
+    event.raw.topics[0] === '0xe1434e25d6611e0db941968fdc97811c982ac1602e951637d206f5fdda9dd8f1'
+  ) {
+    const rawAddress = event.raw.data === '0x' ? event.raw.topics[1] : event.raw.data
+    const address = '0x' + rawAddress.slice(26)
+    event.event = 'ValidatorRemoved'
+    event.returnValues.validator = address
+  }
+}
+
+const processValidatorsEvents = events => {
+  const validatorList = new Set()
+  events.forEach(event => {
+    parseValidatorEvent(event)
+
+    if (event.event === 'ValidatorAdded') {
+      validatorList.add(event.returnValues.validator)
+    } else if (event.event === 'ValidatorRemoved') {
+      validatorList.delete(event.returnValues.validator)
+    }
+  })
+
+  return Array.from(validatorList)
+}
+
+const tryCall = async (method, fallbackValue) => {
+  try {
+    return await method.call()
+  } catch (e) {
+    return fallbackValue
+  }
+}
+
+const getDeployedAtBlock = async contract => tryCall(contract.methods.deployedAtBlock(), 0)
+
+const getPastEvents = async (contract, { event, fromBlock, toBlock, options }) => {
+  let events
+  try {
+    events = await contract.getPastEvents(event || 'allEvents', {
+      options: options || {},
+      fromBlock,
+      toBlock: toBlock || 'latest'
+    })
+  } catch (e) {
+    if (e.message.includes('query returned more than 1000 results') && Number.isInteger(toBlock)) {
```

Comment:
```suggestion
    if (e.message.includes('query returned more than 10000 results')) {
```

While testing this with the monitor, I found that Infura error for max result was updated to `10000`, see [medium post](https://blog.infura.io/infura-june-2019-update-9528c388a3b4)  (FYI @akolotov ).

Also, the check `Number.isInteger(toBlock)` was not present in the previous implementation of this code. Is it really necessary? From the code and manual testing the monitor,  it looks that `toBlock` is a BigNumber and not an integer and is not getting inside this `if` when the max result error is present.

You can use env variables for xDai bridge to test this
```
HOME_RPC_URL=https://dai.poa.network
FOREIGN_RPC_URL=https://mainnet.infura.io/v3/...
HOME_BRIDGE_ADDRESS=0x7301CFA0e1756B71869E93d4e4Dca5c7d0eb0AA6
FOREIGN_BRIDGE_ADDRESS=0x4aa42145Aa6Ebf72e164C9bBC74fbD3788045016
HOME_DEPLOYMENT_BLOCK=759
FOREIGN_DEPLOYMENT_BLOCK=6478417
```

Reply:
> While testing this with the monitor, I found that Infura error for max result was updated to 10000, see medium post (FYI @akolotov ).

Very nice! Thanks!

Reply:
I changed the statement to
`e.message.includes('query returned more than')`
which should continue working even if the exact number is changed again by infura.

As for the check for `Number.isInteger(toBlock)`, the idea was to prevent from getting inside this `if` in a situation where `toBlock == 'latest'`, because the calculations for middleBlock would not work.
I corrected this, and also removed repetition by specifying default parameters in the method's signature.

Reply:
> I changed the statement to
e.message.includes('query returned more than')
which should continue working even if the exact number is changed again by infura.

Thanks! That makes sense

> As for the check for Number.isInteger(toBlock), the idea was to prevent from getting inside this if in a situation where toBlock == 'latest', because the calculations for middleBlock would not work.
I corrected this, and also removed repetition by specifying default parameters in the method's signature.

Thanks for the explanation